frontend/model/contracts/chatroom.js

```diff
@@ -222,11 +231,11 @@ sbp('chelonia/defineContract', {
           notificationType,
           notificationType === MESSAGE_NOTIFICATIONS.ADD_MEMBER ? { memberID, actorID: innerSigningContractID } : { memberID }
         )
-        const newMessage = createMessage({ meta, hash, height, data: notificationData, state, innerSigningContractID })
-        state.messages.push(newMessage)
+        state.messages.push(createMessage({ meta, hash, height, state, data: notificationData, innerSigningContractID }))
+
+        cutMessagesIfOverflow(state)
```

Comment:
I think this should be DRY'd by doing something like:

```js
addMessage(state, createMessage({ meta, hash, height, state, data: notificationData, innerSigningContractID }))
```

This way it's 1-line instead of 2 lines repeated everywhere. In other words, rename `cutMessagesIfOverflow` to `addMessage` and do the pushing and checking in there.

Reply:
Good idea!